src/utils.js

```diff
@@ -23,6 +24,22 @@ export function isPromise (value) {
   return !!value && typeof value.then === 'function'
 }
 
+export function isOnline () {
+  if (navigator === undefined) {
+    return undefined
+  }
+
+  return navigator.onLine === true
+}
+
+export function isOffline () {
```

Comment:
Regarding the code duplication and the needs, don't you think it is too much code.

```js
export function isOnline () {
  return navigator ? navigator.onLine : undefined
}

export function isOffline () {
  return !isOnline()
}
```

Or with even less code, you define only one function and the developers could use `isOnline()` or `!isOnline()` instead of `isOnline()` and `isOffline()`.